kolibri/core/logger/api.py

```diff
@@ -90,6 +96,21 @@ def update(self, request, *args, **kwargs):
                 default_response[field] = method(instance)
         return Response(default_response)
 
+    def create(self, request, *args, **kwargs):
+        try:
+            return super(LoggerViewSet, self).create(request, *args, **kwargs)
+        except IntegrityError:
+            # The object has been created previously: let's calculate its id and return it
+            serializer = self.get_serializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+            obj = serializer.Meta.model(**serializer.validated_data)
+            obj.id = obj.calculate_uuid()
+            final_obj = self.get_serializer(obj)
+            return Response(final_obj.data)
+        except Exception as e:
+            logger.error("Failed to validate data: {}".format(e))
```

Comment:
Would it work to just reraise the exception here, in case it is something more severe than just a validation error?

Reply:
yep, good idea. I only could reproduce the validation error (`serializer.is_valid(raise_exception=True) ` is what does it) but that does not exclude other possibilities. I will add it.

Reply:
Done